src/schema/users.ts

```diff
@@ -1044,6 +1078,55 @@ export const resolvers: IResolvers<any, Context> = {
 
       return { _: true };
     },
+    acceptFeatureInvite: async (
+      _,
+      args: {
+        token: string;
+        referrerId: string;
+        feature: string;
+      },
+      ctx: Context,
+    ): Promise<unknown> => {
+      const featureRepo = ctx.con.getRepository(Feature);
+      const existingFeature = await featureRepo.findOneBy({
+        userId: ctx.userId,
+        feature: args.feature as FeatureType,
+      });
+
+      // silently fail if we already have the feature
+      if (existingFeature?.value === FeatureValue.Allow) return undefined;
+
+      const inviteRepo = ctx.con.getRepository(Invite);
+      const referrerInvite = await inviteRepo.findOneByOrFail({
+        userId: args.referrerId,
+        token: args.token,
+        campaign: args.feature as CampaignType,
+      });
+
+      if (referrerInvite.count >= referrerInvite.limit) {
+        throw new ValidationError('Invites limit reached');
+      }
+
+      if (existingFeature) {
+        existingFeature.invitedById = args.referrerId;
+        existingFeature.value = FeatureValue.Allow;
+        await featureRepo.save(existingFeature);
+      } else {
+        await featureRepo.save({
+          userId: ctx.userId,
+          feature: args.feature as FeatureType,
+          invitedById: args.referrerId,
+          value: FeatureValue.Allow,
+        });
+      }
+
+      await inviteRepo.update(
```

Comment:
This should happen in a transaction together with the insert statement above. We want both of them to fail or succeed as one atom